packages/vue/src/components/molecules/SfGallery/SfGallery.js

```diff
@@ -30,54 +30,94 @@ export default {
       type: Object,
       default() {
         return {
-          type: "slider",
+          type: 'slider',
           autoplay: false,
           rewind: false,
-          gap: 0
+          gap: 0,
         };
-      }
+      },
+    },
+    /**
+     * Image zoom inside or overlap the stage
+     */
+    inside: {
+      type: Boolean,
+      default: false,
     }
   },
 
   data() {
     return {
+      positionStatic: '',
+      eventHover: '',
+      pictureSelected: '',
       glide: null,
-      activeIndex: this.current - 1
+      activeIndex: this.current - 1,
     };
   },
 
   computed: {
     mapPictures() {
       // map images to handle picture tags with SfImage
-      return this.images.map(({ normal, big }) => {
-        return {
-          small: normal,
-          normal: big
-        };
-      });
-    }
+      return this.images.map(({ normal, big }) => ({
+        small: normal,
+        normal: big,
+      }));
+    },
   },
 
   methods: {
+    positionObject(index) {
+      if (this.$refs.sfGalleryBigImage) {
+        if(this.inside){
+          return this.$refs.sfGalleryBigImage[index].$el.getBoundingClientRect();
+        }else{
+          return this.$refs.sfGalleryBigImage[index].$refs.imgLazy.getBoundingClientRect();
+        }
+      }
+      return '';
+    },
     go(index) {
       this.activeIndex = index;
       /**
        * Event for current image change (`v-model`)
        * @type {Event}
        */
-      this.$emit("click", index + 1);
+      this.$emit('click', index + 1);
       if (this.glide) {
-        this.glide.go("=" + index);
+        this.glide.go(`=${index}`);
       }
-    }
+    },
+    addImg(picture) {
+      this.pictureSelected = picture.normal.url;
```

Comment:
i think you're selecting img for zoom here, right?
should try `zoom` img first, then `big`, and then `normal`, by doing that you'll use biggest possible image for zoom effect.

```suggestion
      const { zoom, big, normal } = picture
      this.pictureSelected = (zoom || big || normal).url
```